### PR TITLE
fix: support for yaml file references

### PR DESCRIPTION
**Pull request by johanneswuerbach**
https://github.com/omissis/go-jsonschema/pull/153 broke the support for yaml references.

This PR fixes it and adds a tests.

### CODE REVIEW

File: pkg/generator/schema_generator.go [hunk 1]
```diff
@@ -92,11 +91,6 @@ func (g *schemaGenerator) generateReferencedType(ref string) (codegen.Type, erro
 	sg := g
 
 	if fileName != "" {
-		fileExt := filepath.Ext(fileName)
-		if fileExt != ".json" {
-			return nil, fmt.Errorf("%w: '%s', only json schema files are allowed", ErrUnsupportedRefExtension, fileExt)
-		}
```
Comment (johanneswuerbach) [inline pkg/generator/schema_generator.go:98]: The call to `g.fileLoader.Load` is already checking for supported file extensions https://github.com/omissis/go-jsonschema/blob/main/pkg/schemas/loaders.go#L143-L149, hence it seems unnecessary to perform this check here again.

**Comment by omissis** [comment]
hey @johanneswuerbach , thanks for the PR, I will check it as soon as I can (probably during the xmas holidays) 🙏

**Review by omissis — APPROVED** [review]

**Comment by johanneswuerbach** [comment]
@omissis any chance we could get a new release with this PR and the latest main?

**Comment by omissis** [comment]
@johanneswuerbach I guess that could be done :)

**Comment by omissis** [comment]
@johanneswuerbach et voilà, fresh out of the oven -> https://github.com/omissis/go-jsonschema/releases/tag/v0.15.0

**Pull request merged**